### DISCUSSION TITLE
Jak duże będą te projekty?

### DISCUSSION THREAD

**Discussion by mlodyoskar** [Frontlive/Start-Coding#3] [Pomysły]
W trakcie stream'a jak dobrze pamiętam pokazywaliście, że przykładowy projekt to byłaby formularz z dynamiczną ilością input'ów. Powiedzmy sobie umownie, że to mały projekt. Czy przewidujecie, żeby projekty były większe? Np., Coś pokroju napisania całej aplikacji do zarządzania treningami. Wtedy pewnie ciężej byłoby, żeby dołączyć do tego design, ale myślę, że jednak z budowania większych aplikacji można się więcej nauczyć.

ps. Co powiecie żeby projekt nazywał się np.  _jakiprojektzrobic.pl_ czy coś w tym stylu, żeby ta nazwa mega sugerowała, o co chodzi i często właśnie w tej formie pada pytanie na discordzie, a myślę, że można by też zrobić do tego jakąś komendę na bota discord'owego i tym odpowiadać. wdyt?

**Comment by Survikrowa** [comment]
Siema. Na streamie wspominaliśmy, że planujemy tutaj różnej wielkości projekty, tak więc bez obaw pojawią się!
Co do nazwy jest ona mocno WIP. Nad domeną i logiem będziemy sie zastanawiać raczej w późniejszych etapach :D Ale twoja propozycja jest naprawdę niezła

**Comment by stepaniukm** [comment]
Myślę, że nic nie stoi na przeszkodzie, żeby projekty były też większe, wszystko właśnie zależy od poziomu. Tylko też to nie jest takie proste, bo chodzi o to, żeby te zadanka były w miarę pozbawione zależności. Głównym targetem są na ten moment frontend developerzy. Żeby zrobić większy projekt jednak potrzeba mieć już jakiegoś rodzaju infrastrukturę backendową i trzeba pomyśleć jak to ograć. Na pewno nie można zakładać, że ktoś się pojawi i to napisze, bo to nie jest takie hop siup. Są narzędzia w stylu [Postgraphile](https://www.graphile.org/postgraphile/introduction/) albo [Hasura](https://hasura.io/), które byłby w stanie pomóc, ale są one bardzo proste i nie oferują nic więcej poza CRUD dla takiego potencjalnego użytkownika. Nadal byłby potrzebny ktoś kto by dopisał trochę logiki biznesowej a to kosztuje czas.

**Reply by mlodyoskar** [reply]
Rozumiem, ma to sens jak najbardziej że ta złożoność nie może być za dużo skoro targetem są frontend developerzy. Choć jeśli taki projekt nie wychodziłby poza nic więcej jak właśnie odpytywanie API z możliwością wprowadzania danych do bazy i logowania użytkownika to Supabase sprawdziłby się do tego całkiem dobrze, a wiedza backendowa potrzebna do tego, żeby z tego narzędzia korzystać jest niewielka, co sądzicie? 

I dzięki za dwa linki, muszę wypróbować te narzędzia 😁

**Reply by stepaniukm** [reply]
Jak najbardziej, przemyślimy jeszcze temat, Supabase jest bardzo fajna i warto umieć z niej korzystać, ale dalej pozostaje temat tego, że ktoś musi tę logikę napisać :D Myślę, że na początek damy po prostu możliwość w ramach naszej aplikacji zdefiniowania adresu url backendu przy definiowaniu zadania, także piłeczka będzie po stronie osoby tworzącej zadanie, żeby zapewnić ewentualną komunikację, jednak nic więcej nie jesteśmy w stanie na ten moment raczej zrobić.

**Reply by stepaniukm** [reply]
@mlodyoskar Zrobiłem na to #5 issue także tam możemy pilnować tego wszystkiego co się dowiemy w tym temacie.
